AudioRecorder.py

```diff
@@ -1,7 +1,15 @@
 import custom_speech_recognition as sr
-import pyaudiowpatch as pyaudio
+import os
 from datetime import datetime
 
+try:
+    import pyaudiowpatch as pyaudio
+except ImportError:
+    if os.name != "nt":
```

Comment:
1. Would like a design where we only query the OS once, and inject the OS dependency  into the generic code.

Reply:
I agree with you that a structure like this:
```python
if os.name != "nt":
    import pyaudiowpatch as pyaudio
else:
    import pyaudio
```

would look better. But since now the main platform is Windows, I thought that doing a check every time on an OS that is not fully supported would be redundant. And I thought the approach I chose was the best.

However, I don't mind changing that.